Mathlib/Computability/DFA.lean

```diff
@@ -247,3 +245,7 @@ theorem comap_reindex (f : α' → α) (g : σ ≃ σ') :
 end Maps
 
 end DFA
+
+/-- A regular language is a language that is defined by a DFA with finite states. -/
+def Language.IsRegular {T : Type u} (L : Language T) : Prop :=
+  ∃ σ : Type v, ∃ _ : Fintype σ, ∃ M : DFA T σ, M.accepts = L
```

Comment:
This is a bad definition, in that the universe variable `v` only shows up in the body of the definition and not in the arguments. This is problematic as you then have countably many predicate `IsRegular.{0}`, `IsRegular.{1}`, `IsRegular.{2}`, ... which are all equivalent.

Please do
```suggestion
/-- A regular language is a language that is defined by a DFA with finite states. -/
def Language.IsRegular {T : Type u} (L : Language T) : Prop :=
  ∃ σ : Type, ∃ _ : Fintype σ, ∃ M : DFA T σ, M.accepts = L

lemma Language.isRegular_iff {T : Type u} {L : Language T}
    L.IsRegular ↔ ∃ σ : Type v, ∃ _ : Fintype σ, ∃ M : DFA T σ, M.accepts = L := sorry
```
which will involve using `Shrink` and importing `Mathlib.Data.Countable.Small`

Reply:
Unfortunately, I don't understand how `Shrink` works, so let's keep it at `proof_wanted` in this PR.